src/models/nn/build_nn.cc

```diff
@@ -507,11 +514,11 @@ struct build_nn_fn {
 
     NNContext nn_context(context, coefficients, bias);
 
-    Matrix<T> X      = Matrix<T>::Project3dStore(X_store, 2);
-    Matrix<double> g = Matrix<double>::Project3dStore(g_store, 1);
-    Matrix<double> h = Matrix<double>::Project3dStore(h_store, 1);
+    auto X = Matrix<const T>::Project3dStore(X_store, 2);
+    auto g = Matrix<const double>::Project3dStore(g_store, 1);
+    auto h = Matrix<const double>::Project3dStore(h_store, 1);
 
-    std::vector<Matrix<T>> activations({X});
+    std::vector<Matrix<T>> activations;
```

Comment:
Only comment would be if this is obvious enough, or whether it makes sense to point out once that `acivations` doesn't include the first entry (as a code comment).  Not sure that would be here.